libraries/chain/trace.cpp

```diff
@@ -34,4 +35,29 @@ action_trace::action_trace(
 ,producer_block_id( trace.producer_block_id )
 {}
 
+ram_trace::ram_trace(
+   uint32_t action_id,
+   const char* event_id,
+   const char* family,
+   const char* operation,
+   const char* legacy_tag
+)
+:action_id( action_id )
+,event_id( event_id )
+,family( family )
+,operation( operation )
+,legacy_tag( legacy_tag )
+{
+   EOS_ASSERT(!is_generic(), misc_exception, "ram trace from constructor cannot be generic");
```

Comment:
The idea was indeed to prevent programmer error that would see a `ram_trace` param and would try to construct an empty `ram_trace`, with default values like `ram_trace(0, nullptr, nullptr, nullptr, nullptr)`. That being said, there is probably a bunch of other possibilities someone would use here like empty strings, and the assertions could not cover all cases.

The idea behind `generic_ram_trace` was to improve readability, usage and searchability (for me). Like if you don't know how to construct a full `ram_trace`, use a `generic_ram_trace` instead.

That's what I kept for now, but I could switch back to a single `ram_trace` struct.